app/javascript/shared/serviceWorker.js

```diff
@@ -0,0 +1,17 @@
+window.addEventListener('load', () => {
+  navigator.serviceWorker.register('/service-worker.js').then(registration => {
+    var serviceWorker;
+    if (registration.installing) {
+      serviceWorker = registration.installing;
+      console.log('Service worker installing.');
+    } else if (registration.waiting) {
+      serviceWorker = registration.waiting;
+      console.log('Service worker installed & waiting.');
+    } else if (registration.active) {
+      serviceWorker = registration.active;
+      console.log('Service worker active.');
+    }
```

Comment:
What exactly is this chunk of code doing? It's defining a local variable `serviceWorker`, and then assigning a value to it. This value isn't then used anywhere. What's the purpose of the assignments in the conditional blocks?

Also, are the debug log commands still required?